src/arrayMethodSort.js

```diff
@@ -4,8 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  const createdSortFunc = (a, b) => {
+    const stringA = String(a);
+    const stringB = String(b);
+
+    if (stringA > stringB) {
+      return 1;
+    }
+
+    if (stringA < stringB) {
+      return -1;
+    }
+
+    return 0;
+  };
+
+  [].__proto__.sort2 = function(compareFunction = createdSortFunc) {
+    let count = 0;
+
+    do {
+      count = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        if (compareFunction(this[i - 1], this[i]) > 0) {
+          const placeholder = this[i];
+
+          this[i] = this[i - 1];
+          this[i - 1] = placeholder;
```

Comment:
This sections gets a bit confusing, `this[i]` and `this[i - 1]` are used more than once, and the variable `const placeholder = this[i]` is also created. For readability, perhaps using variables for both  `this[i]` and `this[i - 1]` in this loop would be a better idea.

Reply:
created the variables